src/train_and_predict_model.py

```diff
@@ -1,28 +1,34 @@
-#' train model and predict based on a specified algorithm
-#'
-#' @param string the name of algorithm will be used
-#' @param integer the number of K
-#' @param *array sequence of opriginal results in test set
-#' @param *array sequence of predicted results in test set
-#'
-#' @return an array of accuracies in terms of different Ks based on a specified algorithm
-#'
-#' @export
-#'
-#' @examples
-#' TrainAndPredictModel('KNN', 81, X_train, y_train, y_test)
-def train_and_predict_model(algorithm, Ks, X_train, y_train, y_test):
+from sklearn import metrics
+from sklearn.neighbors import KNeighborsClassifier
+from sklearn.tree import DecisionTreeClassifier
+
+
+# ' train model and predict based on a specified algorithm
+# '
+# ' @param string, the name of algorithm will be used
+# ' @param integer, the number of neighbors of KNN or the depth of Decision Tree
+# ' @param *array, sequence of training set
+# ' @param *array, sequence of test set
+# ' @param *array, sequence of observed results in training set
+# ' @param *array, sequence of observed results in test set
+# ' @param *array, an (Ks-1)-long array being used to accommodate accuracy score
+# '
+# ' @return an array of accuracies in terms of different Ks based on a specified algorithm
+# '
+# ' @export
+# '
+# ' @examples
+# ' TrainAndPredictModel('KNN', 81, X_train, y_train, y_test, mean_acc)
+def train_and_predict_model(algorithm, Ks, X_train, X_test, y_train, y_test, mean_acc):
+
     for n in range(1, Ks):
         if algorithm == 'DT':
-            decTree = DecisionTreeClassifier(criterion="entropy", max_depth = n)
-            decTree.fit(X_train,y_train)
-            yhat=decTree.predict(X_test)
+            decTree = DecisionTreeClassifier(criterion="entropy", max_depth=n)
+            decTree.fit(X_train, y_train)
+            yhat = decTree.predict(X_test)
+            mean_acc[n - 1] = metrics.accuracy_score(y_test, yhat)
         elif algorithm == 'KNN':
```

Comment:
advancement not required: probably trimming and checking case sensitive less ly
```suggestion
        elif algorithm.lower().strip() == "knn":
```